File: NCore/Commands/Commandkick.cs
using System.Linq;
using System.Threading.Tasks;
using NCore;
using NCore.netcraft.server.api;

namespace NCore
{
    public class Commandkick : Command
    {
        public Commandkick() : base("kick", NCore.GetNCore().lang.get("commands.kick.description"), "kick <игрок> [причина]")
        {
        }

        public override async Task<bool> OnCommand(CommandSender sender, Command cmd, string[] args, string label)
        {
            NCore.Lang lang = sender.IsPlayer ? ((NetcraftPlayer)sender).lang : NCore.GetNCore().lang;
            if (!sender.GetAdmin())
            {
                await sender.SendMessage(lang.get("commands.generic.error.no-perms"));
                return true;
            }

            if (args.Length == 1)
            {
                var p = Netcraft.GetPlayer(args[0]);
                if (NCore.IsNothing(p))
                {
                    await sender.SendMessage(lang.get("commands.generic.player.not-found"));
                    return true;
                }

                p.Kick(lang.get("commands.kick.kicked"));
                await Netcraft.Broadcast(lang.get("commands.kick.success", sender.GetName(), p.Username, "None"));
                return true;
            }
            else if (args.Length > 1)
            {
                var p = Netcraft.GetPlayer(args[0]);
                if (NCore.IsNothing(p))
                {
                    sender.SendMessage(lang.get("commands.generic.player.not-found"));
                    return true;
                }

                p.Kick(p.lang.get("commands.kick.kicked.reason", string.Join(" ", args.Skip(1))));
                await Netcraft.Broadcast(lang.get("commands.kick.success", sender.GetName(), p.Username, string.Join(" ", args.Skip(1))));

                return true;
            }

            return false;
        }
    }
}

File: NetcraftServer/Commands/Commandkick.vb
Imports NetcraftServer

Public Class Commandkick
    Inherits Command
    Sub New()
        MyBase.New("kick", "Выгоняет игрока с сервера", "kick <игрок> [причина]")
    End Sub
    Public Overrides Function OnCommand(sender As CommandSender, cmd As Command, args() As String, label As String) As Boolean
        If Not sender.GetAdmin Then
            sender.SendMessage("У Вас недостаточно прав!")
            Return True
        End If
        If args.Length = 1 Then
            Dim p = netcraft.server.api.NCSApi.Netcraft.GetPlayer(args(0))
            If IsNothing(p) Then
                sender.SendMessage("Игрок не найден!")
                Return True
            End If
            p.Kick("Вы были выгнаны администратором.")
            netcraft.server.api.NCSApi.Netcraft.Broadcast($"{sender.GetName} выгнал игрока {p.GetName}. Причина не указана.")
        ElseIf args.Length > 1 Then
            Dim p = netcraft.server.api.NCSApi.Netcraft.GetPlayer(args(0))
            If IsNothing(p) Then
                sender.SendMessage("Игрок не найден!")
                Return True
            End If
            p.Kick("Вы были выгнаны администратором. Причина: " + String.Join(" ", args.Skip(1)))
            netcraft.server.api.NCSApi.Netcraft.Broadcast($"{sender.GetName} выгнал игрока {p.GetName}. Причина: {String.Join(" ", args.Skip(1))}")
        End If
        Return False
    End Function
End Class
